articles/quasar-google-auth-with-firebase/index.md

```diff
@@ -0,0 +1,640 @@
+
+```JavaScript
+import firebase from "firebase";
+
+const firebaseConfig = {
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+  firebase.initializeApp(firebaseConfig);
+
+export default firebase
+```
+
+### Access Control for Routes
+
+We're going to disable access to routes which requires authorization when a user needs to access them. This is quite easy. Navigate to `routes` folder, open `index.js` file.
+
+`Import firebase from "firebase"` at the top of the file just like we did in the `firebase.js` boot file. When done, add the following code just above the line `return Router`:
+
+```JavaScript
+  Router.beforeEach(async (to, from, next) => {
+    const auth = to.meta.requiresAuth
+    if (auth && !await firebase.getCurrentUser()) {
+      next('/');
+    } else {
+      next();
+    }
+  })
+```
+
+Now if the currentUser is null or undefined, we should redirect users to the auth path (`/`). But how do we get currentUser? We can’t use `firebase.auth().currentUser` because on page refresh that property has not been set yet before the requiresAuth guard is triggered. 
+
+We will have to use the onAuthStateChanged callback somehow. We have to add a method to the firebase object after we initialize the firebase app. The method is added to `firebase.js` boot file. Edit the firebase boot file to resemble the code below.
+
+```JavaScript
+const firebaseConfig = {
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+  firebase.initializeApp(firebaseConfig);
+
+  firebase.getCurrentUser = () => {
+    return new Promise((resolve, reject) => {
+      const unsubscribe = firebase.auth().onAuthStateChanged(user => {
+        unsubscribe();
+        resolve(user);
+      }, reject);
+    })
+  };
+```  
+
+The method after `firebase.initializeApp(firebaseConfig);` will return a Promise which resolves currentUser as soon as it is set. `onAuthStateChanged` will trigger the callback immediately with either null or the user object if signed in. Then we unsubscribe to not listen for further changes.
+
+### Activate SignIn Methods
+
+We need to activate signin methods provided by firebase, we'll activate the `google` and `Email/Password` providers. Go back to the firebase console where we copied the firebase SDK script. click `Continue to console`, it will take you to the project overview.
+
+On the left sidebar click `Authentication`, then click `Set up sign-in method`. Click on `Email/Password` and `Google` providers to activate them. For Email/Password click the first toggle button, don't enable the `passwordless sigin-in` option, while for the Google provider make sure to fill in your `Project support email ` preferably your email address.
+
+Once done, we can get started adding functionalities on our view.
```

Comment:
Can you provide helpful screenshots for these actions? It'll be easier for the readers to follow along. You can highlight the buttons that the user should click to make it even more easier for the reader to follow.